src/openassetio-python/module/CMakeLists.txt

```diff
@@ -159,15 +154,21 @@ if (Python_Interpreter_FOUND)
         --no-index --find-links ${python_deps_dir}
         ${PROJECT_SOURCE_DIR}
     )
-    add_dependencies(openassetio-python-py-install
```

Comment:
[fix] Removing these dependencies breaks using these targets as part of a build process, as the only dependencies are introduced by `ctest`. Eg: `cmake --build build --target openasetio-python.install` no longer works on a clean build, if you just wanted a python install complete with cmodule, etc:
```
-- Installing OpenAssetIO core Python component
/bin/sh: /Users/tom/dev/OpenAssetIO/build/dist/bin/python: No such file or directory
```
  I think we need to keep this working as a general build system with dependencies between components properly tracked, without having to use tests as the trigger.

Reply:
Capturing off-line discussions - can we have some kind of prefix/suffix that marks these internal leaf tasks as "do not use" for people inspecting the list of available cmake targets?